### PR TITLE
feat: Enable Rule CA1854

### PR DESCRIPTION
**Pull request by workgroupengineering**
## What does the pull request do?

Avoid multiple loopbak diction [CA1854](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1854)

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Fixed issues

Part of #8944

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0034414-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Review by maxkatz6 — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0034416-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

File: src/Avalonia.Diagnostics/Diagnostics/Views/MainWindow.xaml.cs [hunk 1]
```diff
@@ -237,9 +237,9 @@ private void RawKeyDown(RawKeyEventArgs e)
                             else
                             {
                                 //TODO Use Dictionary.Remove(Key, out Value) in netstandard 2.1
-                                if (_frozenPopupStates.ContainsKey(popup))
+                                if (_frozenPopupStates.TryGetValue(popup,out var value))
```
Comment (robloo) [inline src/Avalonia.Diagnostics/Diagnostics/Views/MainWindow.xaml.cs:240]: Nit: missing space after comma

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0034449-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Pull request merged**